R/FilterSpectroOut.R

```diff
@@ -0,0 +1,659 @@
+        }
+        return(x)
+    })
+
+    ## filter NAs
+    if(tolower(filterNA)!="none"){
+        if(length(quantityList)==1){
+            quantityList <- quantityList[[1]]
+        }
+        quantityList <- filterNAsFromList(quantityList, filterNA, annotS,
+                                          infoCondition, maxNAs)
+    }
+
+
+    if(is.null(quantityList)){
+        quantityList <- unlist(quantityList)
+    }
+
+    return(quantityList)
+}
+
+#' @title Filtering data for running FHTHjoin mode
+#'
+#' @description Function for preprocessing data before fitting models.
+#' Filtering can be applied based on different features such as digest type,
+#' proteotypic or number of missed cleavages.
+#'
+#' @param quantityList A list of matrices, containing quantities of interest
+#' (e.g. peptide, modified peptide, precursor) and protein abundances. Rows
+#' represent features and columns samples. Sample names between the different
+#' matrices contained in the list should match. Not matching samples will be
+#' removed.
+#' Output from \code{extractMSData} or \code{extractSpectroData} is in
+#' the correct format (except of potentially not matching sample names, then
+#' these need to be adjusted manually.)
+#' \code{quantityList} myst include the following matrices
+#' \itemize{
+#'   \item 'LiPPep': LiP peptide quantities (or alternatively the modified
+#'   peptide/ precursor/other quantities, dependent on \code{quantName} and
+#'   \code{quantValue})
+#'   \item 'TrPPep': TrP peptide quantities (or alternatively the modified
+#'   peptide/precursor/other quantities, dependent on \code{quantName} and
+#'   \code{quantValue})
+#'   \item 'TrPProt': TrP protein quantities
+#'   }
+#' @param annotPP A data.frame with peptides (/modified peptides/precursors) and
+#' protein annotation. Rows are features and should be the same identifiers as
+#' in the row names of the \code{quantityList}. Data.frame must contain columns
+#' providing the \code{nameIDQuant} and matching protein names in
+#' \code{nameProtQuant}.
+#' @param nameIDQuant A character string giving column of \code{annotPP} which
+#' matches to the row.names of \code{quantityList} or \code{quantityMatrix}.
+#' Default is 'quantID'.
+#' @param nameProtQuant A character string giving column of \code{annotPP} which
+#' protein names are provided. If a peptides matches to the same protein several
+#' times, the protein name should be provided each time, separated by ','. If a
+#' peptide maps to the multiple proteins, these different proteins can be
+#' provided by separating them with ';'.
+#' Default is 'Protein'.
+#'
+#' @return Returns a list of matrices containing full-tryptic and half-tryptic
+#' LiP quantities of interest (e.g. peptide, modified peptide, precursor) and
+#' corresponding TrP protein quantities. The row names in all matrices
+#' correspond to the LiP quantity ID.
+
+filterForFTHTjoin <- function(quantityList, annotPP, nameIDQuant,
```

Comment:
Maybe use the same syntax for defining all functions. Above you set entries =NULL not just put the names if there is no default. But also no idea what the bioconductor guidelines are might be completely wrong here xD